src/Host/Protocol/Impl/CustomHttpError.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Host.Protocol {
+    public enum CustomHttpError {
```

Comment:
My suggestion would be a custom `RTVS-Broker-Error` HTTP header, and we might as well use strings there rather than numbers to make it easier to see what's going on (and convert those strings to/from enums on either side with the usual `Parse` and `ToString`).